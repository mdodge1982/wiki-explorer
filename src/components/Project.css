.Project {
	margin: 0;
}
.Project h4 {
	margin: 0;
	padding: .7em 1em;
	border-bottom: 1px solid #ddd;
	cursor: pointer;
}
.jump-link:before {
	display: block;
	content: '';
	margin-top: -4em;
	height: 4em;
	visibility: hidden;
	pointer-events: none;
}
