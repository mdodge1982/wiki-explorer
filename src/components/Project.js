import React, {Component} from 'react';
import {connect} from 'react-redux';
import {selectProject} from '../actions';
import Pages from './Pages';
import './Project.css';

class ProjectComponent extends Component {
	render() {
		const project = this.props.project;
		const attrs = {};
		let className = 'Project';
		if(this.props.isNewIdx){
			attrs.id = project.name.charAt(0);
			className += ' jump-link';
		}
		let content = '';
		if(project.pages.length>0){
			content = (
				<Pages pages={project.pages} />
			)
		}
		if(project.selected){
			className += ' open';
		}
		if(project.isFetching){
			className += ' red';
		}
		return (
			<li {...attrs} className={className}>
				<h4 onClick={() => this.handleClick(project.name)}>{project.name}</h4>
				{content}
			</li>
		);
	}
	handleClick(projectName) {
		this.props.dispatch(selectProject(projectName));
	}
}

//Use connect to get state and dispatch
const mapStateToProps = ({byName},ownprops) => {
	const project = byName[ownprops.projectName];
	return {project};
}
const Project = connect(mapStateToProps)(ProjectComponent);

export default Project;
