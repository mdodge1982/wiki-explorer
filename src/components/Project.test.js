import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import Project from './Project';
import {Provider} from 'react-redux';
import store from '../store';
import renderer from 'react-test-renderer';

jest.mock('../actions/websocket');

test('Test render', () => {
	store.dispatch({
		type: 'PROJECT.LIST',
		data: ['My Fun Project']
	});
	const component = renderer.create(
		<Provider store={store}>
			<Project projectName="My Fun Project" />
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

});

test('Click selects project', () => {
	const component = ReactTestUtils.renderIntoDocument(
		<Provider store={store}>
			<Project projectName="My Fun Project" />
		</Provider>
	);
	const h4 = ReactTestUtils.findRenderedDOMComponentWithTag(component,'h4');
	ReactTestUtils.Simulate.click(h4);
	expect(store.getState().byName.selected).toEqual('My Fun Project');
});
