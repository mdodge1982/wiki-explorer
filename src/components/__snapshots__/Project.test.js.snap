// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render 1`] = `
<li
  className="Project"
>
  
  <h4
    onClick={[Function]}
  >
    My Fun Project
     
    
  </h4>
  
</li>
`;
